codalab/rest/legacy.py

```diff
@@ -383,6 +326,75 @@ def upload_bundle(self, source_file, bundle_type, worksheet_uuid):
             raise
         return new_bundle_uuid
 
+    # Create an instance of a CLI.
```

Comment:
Did you not read the TODO comments above? You're making this code very confusing.